src/gate_detect/include/Interpolator.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Created By: Cameron Newton
+ * Created On: March 16th, 2019
+ * Description: Class to scan cv::Mat object and return 9 parameter vector
+ * defining where gate is seen on cv::Mat
+ */
+
+#ifndef PROJECT_INTERPOLATOR_H
+#define PROJECT_INTERPOLATOR_H
+
+#include <cmath>
+
+class Interpolator {
+  public:
+    Interpolator(double interpolation_constant1,
+                 double interpolation_constant2);
+    Interpolator();
+
+    enum TYPE { EXPONENTIAL, LINEAR };
+
+    float getVertDistance(int pixel_width);
+
+    float getHorDistance(int pixel_width);
+
+    float getVertAngle(int vertical_mid,
+                       int vert_width,
+                       int width_of_image,
+                       float vertical_distance);
+
+    float getHorAngle(int pixel_height_of_image,
+                      int horizontal_middle,
+                      int horizontal_width,
+                      float horizontal_distance);
```

Comment:
For this function, and for all the other functions we've declared in this PR, we should include some sort of documentation comment. This helps make it much easier for people to read the code and understand what it does in the future. As an example:
```
    /**
     * Get the horizontal angle of the gate from the given image
     * 
     * The angle is measured relative to the vertical angle that the image
     * was taken from
     * @param pixel_height_of_image The height of the image in pixels
     * @param horizontal_middle TODO: not sure what this is
     * @param horizontal_width TODO: not sure what this is
     * @param horizontal_distance The horizontal distance from the gate
     * @return The horizontal angle of the gate, relative to the angle
     *         this image was taken from
     */
    float getHorAngle(int pixel_height_of_image,
                      int horizontal_middle,
                      int horizontal_width,
                      float horizontal_distance);
```

Reply:
Added functions defs